api/src/routers/meals.js

```diff
@@ -0,0 +1,87 @@
+import express from "express";
+import knex from "../database_client.js"; // Import the configured knex instance
+
+const mealsRouter = express.Router();
+
+// GET /api/meals - Returns all meals
+mealsRouter.get("/", async (req, res) => {
+    try {
+        const meals = await knex.select("*").from("Meal").orderBy("id", "ASC");
+        res.json(meals);
+    } catch (error) {
+        res.status(500).json({ error: "Failed to retrieve meals" });
+    }
+});
+
+// POST /api/meals - Adds a new meal to the database
+mealsRouter.post("/", async (req, res) => {
+    const { id, title, description, location, when, max_reservations, price, created_date } =
```

Comment:
probably it would be nice to add some parsing to whether the data coming to the server is good enough to be input into db.

sidenote: is the id here the same as record ID in the db? If so then it would be worth rethinking this and making the db or just backend handle the id'ing :)